CRM/Core/BAO/File.php

```diff
@@ -764,7 +761,7 @@ public static function generateFileHash($entityId = NULL, $fileId = NULL, $genTs
     }
     // Trim 8 chars off the string, make it slightly easier to find
     // but reveals less information from the hash.
-    $cs = hash_hmac('sha256', "file={$fileId}&life={$life}", $siteKey);
+    $cs = hash_hmac('sha256', "file={$fileId}&life={$life}", $signKey);
```

Comment:
Yep, it's a good time to upgrade this to use SIGN_KEYS.

Note that `$signKey` returns the key as an array. The specific property with the secret should be `$signKey['key']`.

Though actually, re-reading, I wonder if we should bite the bullet and just switch it to JWT. The upshot is that we supoprt key-rotation for JWT. (You can phase-in new keys without causing hard breaks on live tokens.)

Something like this (*pseudocode*):

```php
return Civi::service("crypto.jwt")->encode([
  "exp" => $genTs+($days*24*60*60)
  "civi.file" => $fileId,
], ['SIGN', 'WEAK_SIGN']);
```

But then the `validate` function should use `Civi::service("crypto.jwt")->decode()`.